File: BackEnd/Mmo/Mmo_Domain/ModelResponse/UserResponse.cs
namespace Mmo_Domain.ModelResponse;

public class UserResponse
{
    public int Id { get; set; }
    public string Username { get; set; } = null!;
    public string Email { get; set; } = null!;
    public string? Phone { get; set; }
    public decimal? Balance { get; set; }
    public bool? IsActive { get; set; }
    public DateTime? CreatedAt { get; set; }
    public List<string> Roles { get; set; } = new List<string>();
}

File: frontend/src/models/modelResponse/UserResponse.ts
// User Response Models

export interface UpdateProfileResponse {
  id: number;
  username: string;
  email: string;
  phone?: string;
  balance?: number;
  isActive?: boolean;
  createdAt?: string;
  updatedAt?: string;
  avatarUrl?: string;
}

export interface UserForAdmin {
  id: number;
  username: string;
  email: string;
  phone: string;
  balance: number;
  isActive: boolean;
  createdAt: string;
  roles: string[];
}

export interface PaginatedUsersResponse {
  items: UserForAdmin[];
  total: number;
}
